docs/pktoken.md

```diff
@@ -0,0 +1,702 @@
+  {
+    "protected":{
+    "alg": "GQ256",
+    "jkt": "UJCvHZiaJcaQDc2tJbP_kgtgxB-OcKd1lwD76M3riUY",
+    "kid": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6Ikh5cTROQVRBanNucUM3bWRydEFoaHJDUjJfUSIsImtpZCI6IjFGMkFCODM0MDRDMDhFQzlFQTBCQjk5REFFRDAyMTg2QjA5MURCRjQifQ",
+    "typ": "JWT"},
+    "signature":"h4lbgJd... (5504 base64 characters)"
+  },
+  {
+    "protected": {
+    "alg": "ES256",
+    "rz": "bca0353ea63adbfce72032ab7d8fb7940def3488ca0765546a89d46760113c70",
+    "typ": "CIC",
+    "upk": {
+        "alg": "ES256",
+        "crv": "P-256",
+        "kty": "EC",
+        "x": "5BP8B8bXgf0OFxHLJS5LSFlPOsfdIvf2tJU_3mwTGNE",
+        "y": "7KzWJi88qdZOI_j-kUG2aPjkzEA7IGMXFp1f-jdt28I"
+    }},
+    "signature":"5dfaFj..."
+  }
+]}
+```
+
+#### Audience-Commitment
+
+Machine identity ID Token issuance flows typically allow the identity requesting the ID Token to specify any value for the audience `aud`. Most machine identity flows do not use a `nonce`. To support OpenPubkey we use the `aud` in exactly the same way as the `nonce` commitment:
+
+```ascii
+nonce = SHA3(
+  BASE64URL({"alg": "ES256",
+    "rz": "bca0353ea63adbfce72032ab7d8fb7940def3488ca0765546a89d46760113c70",
+    "typ": "CIC",
+    "upk": {
+        "alg": "ES256",
+        "crv": "P-256",
+        "kty": "EC",
+        "x": "5BP8B8bXgf0OFxHLJS5LSFlPOsfdIvf2tJU_3mwTGNE",
+        "y": "7KzWJi88qdZOI_j-kUG2aPjkzEA7IGMXFp1f-jdt28I"
+    }}))
+```
+
+User identity ID Token issuance flows set the `aud` to a unique identifier to scope the ID Token to a particular service or OIDC client. This is done, among other reasons to prevent a malicious service from replaying the ID Tokens it receives from users to impersonate those users to a another service. Allowing the requesting party to specify the `aud` as is done in machine identity would be insecure for user identity. However it is both secure and the primary pattern in machine identity flows.
+
+#### GQ Signatures Are Required For Aud-Commitment PK Tokens
+
+The main use case of machine identity audience-commitment PK Tokens is to create publicly published signatures. As in such cases GQ signatures or ZK proofs should always be used, in such cases we require the use of GQ signatures. This eliminates the risks of accidental misconfigurations where a GQ signature should be used but is not used.
+
+### GQ-Commitment PK Tokens (Gitlab-CI Example)
+
+GQ-commitment PK Tokens are designed for the case where a OP can not support a nonce-commitment or an audience-commitment. Instead the GQ signature itself functions as the binding between the identity's public key (and CIC protected header) and the ID Token. So far we have only encountered one OP which can't support a nonce-commitment or an aud-commitment: gitlab-CI.
+
+**Critical:** GQ-commitment PK Tokens should only be used if an OP can not support nonce or audience-commitment PK Tokens. Never verify a GQ-bound PK Token unless the issuing OP only supports GQ-bound PK Tokens.
```

Comment:
Is this an easy way to catch malicious PK Tokens (if you see a GQ-bound PK Token from an OP that does not only support GQ-bound PK Tokens?)

Reply:
An attacker is unlikely to do this because we will always reject it. It is a good check for misconfigurations